ONSdigital/response-operations-ui
# Username validation message

# What and why?
Response-ops needed to be changed so that users can change their e-mails via the change username page with the following validation rules:

* Username field must not be empty
* Username must not be longer than 255 characters
* Username must not use characters other than lowercase `a` to `z`, `0` to `9`, and special characters `.`, `@`, and `_`. (Previously those three special characters were unable to be used)

# How to test?
On response-ops, try and change a user's username to an invalid username (based on the validation checks above) and make sure that the error messages appear. Also, make sure that valid usernames (including ones with the `.`, `@`, and `_` special characters).

# Trello
[Card](https://trello.com/c/OmKUhX4Y)
